homeassistant/components/swiss_public_transport/sensor.py

```diff
@@ -121,16 +121,22 @@ def __init__(
             manufacturer="Opendata.ch",
             entry_type=DeviceEntryType.SERVICE,
         )
+        self._async_update_attrs()
```

Comment:
This should done in `async_added_to_hass`.

Reply:
Ok, I got it from here: https://github.com/miaucl/ha/blob/5dd7b3f907e30b04057af55a7771a92e589ab5ac/homeassistant/components/airzone_cloud/sensor.py#L157

Reply:
There's no side effect here currently so it could be set in init method too.

Reply:
I dont have a preferance😊